newm-server/src/main/kotlin/io/newm/server/features/distribution/eveara/EvearaDistributionRepositoryImpl.kt

```diff
@@ -0,0 +1,136 @@
+package io.newm.server.features.distribution.eveara
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import io.ktor.client.HttpClient
+import io.ktor.client.call.body
+import io.ktor.client.plugins.ServerResponseException
+import io.ktor.client.request.accept
+import io.ktor.client.request.bearerAuth
+import io.ktor.client.request.get
+import io.ktor.client.request.post
+import io.ktor.client.request.setBody
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+import io.ktor.http.isSuccess
+import io.ktor.server.application.ApplicationEnvironment
+import io.newm.server.features.distribution.DistributionRepository
+import io.newm.server.features.distribution.model.GetGenresResponse
+import io.newm.server.features.song.database.SongEntity
+import io.newm.shared.koin.inject
+import io.newm.shared.ktx.getConfigString
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+class EvearaDistributionRepositoryImpl : DistributionRepository {
```

Comment:
@wlara Can you create a new ticket for finishing out the implementation for the evara distribution service? I'd like to switch my own focus on the minting piece for now.  Documentation is here: https://staging.eveara.com/documentation/

Ping me with any questions.

Reply:
Will do